GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs

```diff
@@ -2,22 +2,34 @@
 using System.Collections.Generic;
 using System.ComponentModel;
 using System.Drawing;
+using System.Threading.Tasks;
 using System.Windows.Forms;
 using GitCommands;
-using GitCommands.Repository;
+using GitCommands.UserRepositoryHistory;
+using Microsoft.VisualStudio.Threading;
 
 namespace GitUI.CommandsDialogs.BrowseDialog
 {
     public partial class FormRecentReposSettings : GitExtensionsForm
     {
+        private IList<Repository> _repositoryHistory;
+
         public FormRecentReposSettings()
             : base(true)
         {
             InitializeComponent();
             Translate();
-            LoadSettings();
-            RefreshRepos();
-            SetComboWidth();
+
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
❓ Why did you use `Run` for this one and `RunAsync` for `FormOpenDirectory`?

📝 Various other notes apply to this one.